app/lib/check_cached_fields.rb

```diff
@@ -132,11 +132,14 @@ def update_cached_field(name, obj, condition, ids, callback, options)
           pg_field_name: options[:pg_field_name],
           recalculate: options[:recalculate],
         }
-        self.delay_for(1.second).update_cached_field_bg(name, obj, ids, callback, index_options)
+        obj_info = {type: obj.class.name, id: obj.id}.to_json
```

Comment:
Question: Why send `obj_info` with `type` and `id`? Are there cases where `obj_info` will contain other keys? If not, I suggest to pass instead two separate parameters `obj_type` (or `obj_class`) and `obj_id`.

Reply:
I reverted this commit as there is a case related to `after_destroy` which means the object not exists 
I'll file a ticket related to this change request to handle all cases